pulsar-client-cpp/include/pulsar/KeyValue.h

```diff
@@ -0,0 +1,75 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#ifndef KEY_VALUE_HPP_
+#define KEY_VALUE_HPP_
+
+#include <map>
+#include <string>
+#include <memory>
+#include "defines.h"
+#include "Schema.h"
+
+namespace pulsar {
+
+class KeyValueImpl;
+
+/**
+ * Use to when the user uses key value schema.
+ */
+class PULSAR_PUBLIC KeyValue {
+   public:
+    KeyValue();
+    /**
+     * Constructor key value, according to keyValueEncodingType, whether key and value be encoded together.
+     *
+     * @param key  key data.
+     * @param value value data.
+     * @param keyValueEncodingType key value encoding type.
+     */
+    KeyValue(const std::string &key, const std::string &value,
+             const KeyValueEncodingType &keyValueEncodingType);
+    /**
+     * Get the key of KeyValue.
+     *
+     * @return character stream for key
+     */
+    std::string getKey() const;
+    /**
+     * Get the value of KeyValue.
+     *
+     * @return character stream for value
+     */
+    std::string getValue() const;
```

Comment:
Return `const std::string&` instead of `std::string`.

You might have referenced the API design of `Message::getDataAsString`, which returns `std::string` because the underlying type is `SharedBuffer`, not a `std::string`. The `getDataAsString` method always creates a `std::string` object as a returned value. Usually if users want to access the data without extra copying, they should use `getData` and `getLength` to access data by the pointer.

```c++
        // Access the value without copying
        const char* ptr = static_cast<const char*>(msg.getData());
        size_t length = msg.getLength();
        for (size_t i = 0; i < length; i++) {
            std::cout << ptr[i];
        }
        std::cout << std::endl;

        // Access the copied value
        std::string s = msg.getDataAsString();
        for (size_t i = 0; i < s.size(); i++) {
            std::cout << s[i];
        }
        std::cout << std::endl;
```

However, for a given `KeyValue` object, `getKey` and `getValue` just access the underlying `std::string` fields.